napari/utils/colormaps/colormap.py

```diff
@@ -63,8 +63,24 @@ def _check_controls(cls, v, values):
                 values['interpolation'] == ColormapInterpolationMode.ZERO
             )
             return np.linspace(0, 1, n_controls)
+        if not np.array_equal(v, sorted(v)):
+            raise ValueError("Coords needs to be sorted in ascending order")
         return v
 
+    @root_validator
+    def check_bound_coords(cls, values):
+        colors = values['colors']
+        controls = values['controls']
+        if controls[0] != 0:
+            controls = np.concatenate(([0], controls))
+            colors = np.concatenate(([colors[0]], colors))
+        if controls[-1] != 1:
+            controls = np.concatenate((controls, [1]))
+            colors = np.concatenate((colors, [colors[-1]]))
+        values['colors'] = colors
+        values['controls'] = controls
```

Comment:
Can you add an additional check here that both of these have the same length and raise an error if not